lumen/ai/views.py

```diff
@@ -83,7 +84,20 @@ async def _render_component(self):
         # store the spec in the cache instead of memory to save tokens
         memory["current_spec"] = self.spec
         try:
-            self.component = type(self.component).from_spec(load_yaml(self.spec))
+            yaml_spec = load_yaml(self.spec)
+
+            if hasattr(self.component, "source") and hasattr(self.component.source, "create_sql_expr_source"):
+                pipeline = self.component
+                tables = yaml_spec["source"]["tables"]
+                old_source = memory["current_source"]
+                old_source.tables = tables
+                pipeline.param.update(
+                    source=old_source,
+                    table=yaml_spec["table"]
+                )
+                self.component = pipeline
+            else:
+                self.component = type(self.component).from_spec(yaml_spec)
```

Comment:
Semi-hacky, but I don't know how to approach.

I also tried `pipeline.source.create_sql_expr_source` for a very, very long time, but I could not figure out why the stored tables in `_connection` were not transferring over... so instead, I just use the old source and manually update the tables + table.